### ISSUE TITLE
Call an event when farmland decays to dirt due to missing hydration

### ISSUE THREAD

**Issue by Wertzui123** [pmmp/PocketMine-MP#5883]
## Description
There should be an event when farmland decays to dirt due to missing hydration.

## Justification
It is currently not possible to cancel this behavior (https://discord.com/channels/373199722573201408/373214753147060235/1126894765410091059), which is really inconvenient for map builders.

## Alternative methods
Allow overriding builtin blocks.

**Comment by Wertzui123** [comment]
If anyone else needs this, there is a workaround for this problem:
```php
$this->getServer()->getWorldManager()->getWorldByName($world)->removeRandomTickedBlock(VanillaBlocks::FARMLAND());
```
Thanks to @JavierLeon9966 for pointing this out!

**Comment by Wertzui123** [comment]
Closing this as it has been implemented in a5d8ef7a6c5ad93a385de8d3be92eef29e812494.

**Comment by dktapps** [comment]
Thanks. Not sure why it didn't get automatically closed.
